nrf_wifi/fw_if/umac_if/src/event.c

```diff
@@ -970,8 +978,40 @@ static enum nrf_wifi_status umac_process_sys_events(struct nrf_wifi_fmac_dev_ctx
 					def_dev_ctx->vif_ctx[mode_event->if_index]->txinjection_mode
 									= true;
 #endif /* CONFIG_NRF700X_RAW_DATA_TX */
+#ifdef CONFIG_NRF700X_PROMISC_DATA_RX
+					def_dev_ctx->vif_ctx[mode_event->if_index]->promisc_mode
+									= false;
+#endif /* CONFIG_NRF700X_PROMISC_DATA_RX */
 				}
 #endif /* CONFIG_NRF700X_RAW_DATA_RX */
+#ifdef CONFIG_NRF700X_PROMISC_DATA_RX
+				else if (mode_event->op_mode == (NRF_WIFI_STA_MODE |
```

Comment:
Instead of having different `else if` cases to check for combinations of `NRF_WIFI_STA_MODE`, `NRF_WIFI_PROMISCUOUS_MODE`, `NRF_WIFI_TX_INJECTION_MODE` would it be better to just check for:

`else if (mode_event->op_mode & NRF_WIFI_STA_MODE == NRF_WIFI_STA_MODE )` and then have checks internally for :

`if (mode_event->op_mode & NRF_WIFI_TX_INJECTION_MODE == NRF_WIFI_TX_INJECTION_MODE)` and,

`if (mode_event->op_mode & NRF_WIFI_PROMISCUOUS_MODE == NRF_WIFI_PROMISCUOUS_MODE )`

separately so that relevant code can be protected by the compile time flags appropriately ?

Reply:
I need to think about this. I will get back to you on this.

Reply:
I have moved the code into specific mode blocks. Kindly check